Makefile.uk

```diff
@@ -71,7 +79,10 @@ CXXINCLUDES-$(CONFIG_LIBCOMPILER_RT) += -I$(LIBCOMPILER_RT_SRC)/include
 # Global flags
 ################################################################################
 
-LIBCOMPILER_RT_SUPPRESS_FLAGS += -Wno-unused-parameter -Wno-builtin-declaration-mismatch
+ifeq ($(CLANG_CHECK), 0)
+LIBCOMPILER_RT_SUPPRESS_FLAGS += -Wno-unused-but-set-variable -Wno-maybe-uninitialized -Wno-builtin-declaration-mismatch
```

Comment:
See above, the flags should be set if `have_gcc` is set to `y`.
Something like
```make
LIBCOMPILER_RT_SUPPRESS_FLAGS-$(call have_gcc) += -Wno-unused-but-set-variable -Wno-maybe-uninitialized -Wno-builtin-declaration-mismatch
```

Reply:
Took me a while to figure out how this syntax works, but yes, it's much cleaner this way. Updated PR.